src/shared_fs.py

```diff
@@ -0,0 +1,154 @@
+#  Copyright 2023 Canonical Ltd.
+#  See LICENSE file for licensing details.
+
+"""Classes and functions to operate on the shared filesystem between the charm and the runners."""
+import shutil
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Iterator
+
+from errors import (
+    CreateSharedFilesystemError,
+    DeleteSharedFilesystemError,
+    SharedFilesystemNotFoundError,
+    SubprocessError,
+)
+from utilities import execute_command
+
+FILESYSTEM_BASE_PATH = Path("/home/ubuntu/runner-fs")
+FILESYSTEM_SIZE = "1M"
+
+
+@dataclass
+class SharedFilesystem:
+    """Shared filesystem between the charm and the runners.
+
+    Attributes:
+        path: The path of the shared filesystem inside the charm.
+        runner_name: The name of the associated runner.
+    Returns:
+        The shared filesystem.
+    """
+
+    path: Path
+    runner_name: str
+
+
+def _get_runner_image_path(runner_name: str) -> Path:
+    """Get the path of the runner image.
+
+    Args:
+        runner_name: The name of the runner.
+
+    Returns:
+        The path of the runner image.
+    """
+    return FILESYSTEM_BASE_PATH / f"{runner_name}.img"
```

Comment:
So the image mounted using loopback is in the same directory as the mount point right ?
Well have something like:
```
runner_1             dir
runner_1.img      file
runner_2            dir
runner_2img      file
....
```
Shouldn't we have a dedicated folder for the images ?